exporters/exporter.go

```diff
@@ -0,0 +1,72 @@
+package exporters
+
+import (
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// ExporterConfig is a generic string which can be deserialized by each individual Exporter.
+type ExporterConfig string
+
+// ExportData is the interface which all data types sent to Exporters should implement
+type ExportData interface {
+	Round() uint64
+}
+
+// BlockExportData is provided to the Exporter on each round.
+type BlockExportData struct {
+	// Block is the block data written to the blockchain.
+	Block bookkeeping.Block
+
+	// Delta contains a list of account changes resulting from the block. Processor plugins may have modify this data.
+	Delta ledgercore.StateDelta
+
+	// Certificate contains voting data that certifies the block. The certificate is non deterministic, a node stops collecting votes once the voting threshold is reached.
+	Certificate agreement.Certificate
+}
+
+// Round returns the round to which the BlockExportData corresponds
+func (blkData *BlockExportData) Round() uint64 {
+	return uint64(blkData.Block.Round())
+}
+
+// ExporterMetadata contains fields relevant to identification and description of plugins.
+type ExporterMetadata struct {
+	Name        string
+	Description string
+	Deprecated  bool
+}
+
+// Exporter defines the interface for plugins
+type Exporter interface {
+	// Metadata associated with each Exporter.
+	Metadata() ExporterMetadata
```

Comment:
I think you can do something like this, then we can use the same interface for the other plugins
```
type PluginMetadata interface {
	// Metadata associated with each Exporter.
	Metadata() ExporterMetadata
}
```

```suggestion
	PluginMetadata
```

Reply:
Yeah I agree, it seems correct to use the same Metadata construct for all of the plugin types. I'll update this in a subsequent PR to make it more generic. It should probably end up being moved to a separate module in that case as well.